amptorch/trainer.py

```diff
@@ -82,59 +94,99 @@ def get_unique_elements(self, training_images):
         return elements
 
     def load_dataset(self):
-        training_images = self.config["dataset"]["raw_data"]
-        # TODO: Scalability when dataset to large to fit into memory
-        if isinstance(training_images, str):
-            training_images = ase.io.read(training_images, ":")
-        del self.config["dataset"]["raw_data"]
-
-        self.elements = self.config["dataset"].get(
-            "elements", self.get_unique_elements(training_images)
-        )
+        if "lmdb_path" in self.config["dataset"]:
+            self.cache = self.config["dataset"].get("cache", "no")
+            if self.cache == "full":
+                self.train_dataset = AtomsLMDBDatasetCache(
+                    self.config["dataset"]["lmdb_path"],
+                )
+            elif self.cache == "partial":
+                self.train_dataset = AtomsLMDBDatasetPartialCache(
+                    self.config["dataset"]["lmdb_path"],
+                )
+            elif self.cache == "no":
+                self.train_dataset = AtomsLMDBDataset(
+                    self.config["dataset"]["lmdb_path"],
+                )
+            else:
+                raise NotImplementedError
```

Comment:
Any chance we can create a parent class for LMDBs that is instantiated with this `if/else` logic instead of having it all here?